support/controlplane-component/defaults_test.go

```diff
@@ -0,0 +1,60 @@
+package controlplanecomponent
+
+import (
+	"testing"
+
+	fuzz "github.com/google/gofuzz"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestComputeResourceHashConsistency(t *testing.T) {
+	g := NewGomegaWithT(t)
+	for range 10 {
+		hashValue := ""
+		for range 100 {
+			secretData, configMapData := generateResources()
+			result, err := computeResourceHash(_resourceKeys(secretData), _resourceKeys(configMapData),
+				func(name string) (*corev1.Secret, error) {
+					return secretData[name], nil
+				},
+				func(name string) (*corev1.ConfigMap, error) {
+					return configMapData[name], nil
+				})
+			g.Expect(err).ToNot(HaveOccurred())
+			if hashValue == "" {
+				hashValue = result
+				break
+			}
+			g.Expect(result).To(Equal(hashValue), "Hash value must remain the same for the same data")
```

Comment:
can you help me grok how this expectation is true?

Reply:
The test is to ensure that the computed hash remains consistent through several calls to computeResourceHash, so we fuzz a set of inputs once, and then compute the hash several times. Every time we compute it, it should remain the same.